app/src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -190,6 +205,39 @@ public void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    /**
+     * Invoked after user clicked on the {@link mPathView} label.
+     */
+    @SuppressLint("InlinedAPI")
+    private void onPathViewClick() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            startActivityForResult(FolderPickerActivity.createIntent(this, mFolder.path, null),
+                FolderPickerActivity.DIRECTORY_REQUEST_CODE);
+            return;
+        }
+
+        /**
+         * Determine the app's private data folder on external storage if present.
+         * e.g. "/storage/abcd-efgh/Android/com.nutomic.syncthinandroid/files"
+         */
+        File initialPath = null;
+        ArrayList<File> externalFilesDir = new ArrayList<>();
+        externalFilesDir.addAll(Arrays.asList(getExternalFilesDirs(null)));
+        externalFilesDir.remove(getExternalFilesDir(null));
+
+        // Display storage access framework directory picker UI.
+        Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE);
+        if (externalFilesDir.size() > 0) {
+            initialPath = externalFilesDir.get(0);
+            Toast.makeText(this, getString(R.string.folder_hint_writeable_initial_folder, initialPath.getAbsolutePath()),
```

Comment:
I intend to improve this by a future PR.
Reason: With the old FolderPickerActivity now only opening on Android < Lollipop, there was a line pointing to the external storage directory "/storage/....-..../Android/com.nutomic.syncthingandroid/files" to help the user find out where he has write access for doing two-way sync.
I absolutely couldn't figure out doing a lot of research how we can start the SAF folder picker UI with an initial folder to open automatically as I would have chosen to open in this special (long) path at default.

Final solution for a follow-up PR:
We precreate a "Share" folder here initially when the app starts for the first time (and once after app upgrade took place) and then the user as two folders automatically precreated available:
a) "device_name_random_device_model" as Send-Only folder pre-configured to point at "/storage/emulated/0/DCIM" (already implemented since years)
b) "device_name_random_Share" as Send-Receive folder pre-configured to point at "/storage/....-..../Android/com.nutomic.syncthingandroid/files/Share"
Then we can get rid of the nasty toast popup reminder.